File: src/test/java/org/hisp/dhis/rules/models/RuleVariableAttributeTest.java
package org.hisp.dhis.rules.models;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.JUnit4;

import java.util.ArrayList;

import static org.junit.Assert.assertEquals;

@RunWith( JUnit4.class )
public class RuleVariableAttributeTest
{
    @Test
    public void createShouldPropagatePropertiesCorrectly()
    {
        RuleVariableAttribute ruleVariableAttribute = RuleVariableAttribute.create(
            "test_variable", "test_attribute", RuleValueType.NUMERIC, true, new ArrayList<>());

        assertEquals( "test_variable" , ruleVariableAttribute.name() );
        assertEquals( "test_attribute" , ruleVariableAttribute.trackedEntityAttribute() );
        assertEquals( RuleValueType.NUMERIC , ruleVariableAttribute.trackedEntityAttributeType() );
    }
}

File: src/test/java/org/hisp/dhis/rules/models/RuleVariableAttributeTest.kt
package org.hisp.dhis.rules.models

import kotlin.test.Test
import kotlin.test.assertEquals

class RuleVariableAttributeTest {
    @Test
    fun createShouldPropagatePropertiesCorrectly() {
        val ruleVariableAttribute = RuleVariableAttribute(
            "test_variable", true, listOf(), "test_attribute", RuleValueType.NUMERIC
        )
        assertEquals("test_variable", ruleVariableAttribute.name)
        assertEquals("test_attribute", ruleVariableAttribute.trackedEntityAttribute)
        assertEquals(RuleValueType.NUMERIC, ruleVariableAttribute.trackedEntityAttributeType)
    }
}
